From 1d138566fbc217b29c387f91675df950d9d550c5 Mon Sep 17 00:00:00 2001
From: Felix Semmler <felix.semmler@aoe.com>
Date: Thu, 24 Mar 2022 16:30:15 +0100
Subject: [BUGFIX] Add url annotation for swagger to get api explorer
 working

---
 src/Explorer/v2/Explorer.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Explorer/v2/Explorer.php b/src/Explorer/v2/Explorer.php
index 9eef719..b830656 100644
--- a/src/Explorer/v2/Explorer.php
+++ b/src/Explorer/v2/Explorer.php
@@ -193,6 +193,7 @@ class Explorer implements iProvideMultiVersionApi
     }
 
     /**
+     * @url GET swagger.json
      * @return stdClass
      */
     public function swagger()
-- 
2.32.0 (Apple Git-132)

